quiche/src/stream.rs

```diff
@@ -642,12 +647,15 @@ impl StreamMap {
 
 /// A QUIC stream.
 #[derive(Default)]
-pub struct Stream {
+pub struct Stream<F = DefaultBufFactory>
+where
+    F: BufFactory,
+{
     /// Receive-side stream buffer.
     pub recv: RecvBuf,
 
     /// Send-side stream buffer.
-    pub send: SendBuf,
+    pub send: SendBuf<F>,
```

Comment:
Why is only SendBuf parameterized with a BufFactory? Seems like RecvBuf should be parameterized as well. I'm also wondering if there are cases where you might want a different BufFactory for send vs. recv...

Reply:
Yes, that would be a follow up PR, RecvBuf is also important